src/Blocks/BlockPiston.cpp

```diff
@@ -214,37 +214,58 @@ bool cBlockPistonHandler::CanPushBlock(
 
 void cBlockPistonHandler::ExtendPiston(Vector3i a_BlockPos, cWorld & a_World)
 {
-	BLOCKTYPE pistonBlock;
-	NIBBLETYPE pistonMeta;
-	a_World.GetBlockTypeMeta(a_BlockPos.x, a_BlockPos.y, a_BlockPos.z, pistonBlock, pistonMeta);
-
-	if (IsExtended(pistonMeta))
 	{
-		// Already extended, bail out
-		return;
+		// Broadcast block action first. Will do nothing if piston cannot in fact push
+
+		BLOCKTYPE pistonBlock;
+		NIBBLETYPE pistonMeta;
+		a_World.GetBlockTypeMeta(a_BlockPos.x, a_BlockPos.y, a_BlockPos.z, pistonBlock, pistonMeta);
+		a_World.BroadcastBlockAction(a_BlockPos, 0, pistonMeta, pistonBlock);
```

Comment:
Should this magic number be replaced with a constant?

Reply:
They've been here since the code was first added, and you would be so heartless as to remove them now? 👁‍🗨 👁‍🗨